modin/experimental/core/execution/native/implementations/omnisci_on_native/partitioning/partition_manager.py

```diff
@@ -255,6 +255,14 @@ def run_exec_plan(cls, plan, index_cols, dtypes, columns):
         assert rb is not None
         at = pyarrow.Table.from_batches([rb])
 
+        if hasattr(curs, "getArrowTable"):
+            at = curs.getArrowTable()
+        else:
+            rb = curs.getArrowRecordBatch()
+            assert rb is not None
+            at = pyarrow.Table.from_batches([rb])
+        assert at is not None
```

Comment:
check out a [few lines above](https://github.com/novichkovg/modin/blob/9c313eeee35132a70c5225965481ed631146f3e5/modin/experimental/core/execution/native/implementations/omnisci_on_native/partitioning/partition_manager.py#L254-L256), it seems that the old logic of conversion through record batches is still there

Reply:
> check out a few lines above, it seems that the old logic of conversion through record batches is still there
This logic is present for backward compatibility, as requested by @ienkovich in OmniSciDb Issue #211.

Reply:
I mean that it's performed unconditionally. As far as I understand the newly added if-statement ensures backward compatibility:
1. If `curs` supports `getArrowTable` then use this method to get the table.
2. Otherwise use the old way through getting a record batch first.

The lines that I linked duplicate the logic from the "else" branch, there is no sense to perform them, right?

Reply:
The code duplication is now removed (d2002798048ea152f6072130732b3b504e800029)